packages/dev/gui/src/2D/controls/textBlock.ts

```diff
@@ -478,29 +495,78 @@ export class TextBlock extends Control {
 
     protected _breakLines(refWidth: number, refHeight: number, context: ICanvasRenderingContext): object[] {
         this._linesTemp.length = 0;
-        const _lines = this.text.split("\n");
+        const _lines = this._textWrapping === TextWrapping.HTML ? this._parseHTMLText(refWidth, refHeight, context) : this.text.split("\n");
 
-        if (this._textWrapping === TextWrapping.Ellipsis) {
-            for (const _line of _lines) {
-                this._linesTemp.push(this._parseLineEllipsis(_line, refWidth, context));
-            }
-        } else if (this._textWrapping === TextWrapping.WordWrap) {
-            for (const _line of _lines) {
-                this._linesTemp.push(...this._parseLineWordWrap(_line, refWidth, context));
-            }
-        } else if (this._textWrapping === TextWrapping.WordWrapEllipsis) {
-            for (const _line of _lines) {
-                this._linesTemp.push(...this._parseLineWordWrapEllipsis(_line, refWidth, refHeight!, context));
-            }
-        } else {
-            for (const _line of _lines) {
-                this._linesTemp.push(this._parseLine(_line, context));
-            }
+        switch (this._textWrapping) {
+            case TextWrapping.WordWrap:
+                for (const _line of _lines) {
+                    this._linesTemp.push(...this._parseLineWordWrap(_line, refWidth, context));
+                }
+                break;
+            case TextWrapping.Ellipsis:
+                for (const _line of _lines) {
+                    this._linesTemp.push(this._parseLineEllipsis(_line, refWidth, context));
+                }
+                break;
+            case TextWrapping.WordWrapEllipsis:
+                for (const _line of _lines) {
+                    this._linesTemp.push(...this._parseLineWordWrapEllipsis(_line, refWidth, refHeight, context));
+                }
+                break;
+            case TextWrapping.HTML:
+            default:
+                for (const _line of _lines) {
+                    this._linesTemp.push(this._parseLine(_line, context));
+                }
+                break;
         }
 
         return this._linesTemp;
     }
 
+    private _htmlElement: Nullable<HTMLElement> = null;
+
+    protected _parseHTMLText(refWidth: number, refHeight: number, context: ICanvasRenderingContext): string[] {
+        const lines = [] as string[];
+        if (!this._htmlElement) {
+            this._htmlElement = document.createElement("div");
+            document.body.appendChild(this._htmlElement);
+        }
+        const htmlElement = this._htmlElement;
+        htmlElement.textContent = this.text;

```

Comment:
[NIT] Should we not keep most of the previous style/attributes assignments in the if block above ?

Reply:
The element can be adjusted using the `adjustWordWrappingHTMLElement` function. So if someone sets the function, adjusts the element and then changes the function (or null it) the state will be not-controlled. I know it is an edge case (or - edge of an edge of an edge case), but I preferred making sure the element is always ready to be rendered. The default behavior is anyhow not to reuse the element, so it will be called every time.